packages/browser-agent-core/features/metrics/instrument/workers-helper.js

```diff
@@ -0,0 +1,82 @@
+export default {workersApiIsSupported, insertSupportMetrics};
+
+/**
+ * True if Web Workers are supported in browser's execution context. Not all browser versions may support certain Workers or options however.
+ * - Warning: service workers are not available on unsecured HTTP sites
+ */
+const workersApiIsSupported = Boolean(self.Worker && self.SharedWorker && self.ServiceWorker);
+
+let origWorker, origSharedWorker, origServiceWorkerCreate;
+/**
+ * Un-instrument support metrics for Workers.
+ * @returns void
+ */
+function resetSupportability() {
+    if (!origWorker) return;    // Worker wasn't changed by this module or was already restored
+    self.Worker = origWorker;
+    self.SharedWorker = origSharedWorker;
+    self.navigator.serviceWorker.register = origServiceWorkerCreate;
+    origWorker = origSharedWorker = origServiceWorkerCreate = undefined;
+}
+
+/**
+ * Injects code to report Web Workers supportability and usage as metrics, replacing the native API in global scope as a side effect.
+ * @param {func} report - handles reporting of supportability metrics to NR1
+ * @returns void
+ */
+function insertSupportMetrics(report) {
+    if (origWorker) return; // ensure we only insert the metrics once
+    if (!workersApiIsSupported) {
```

Comment:
I think this should be fine, there is a chance we would miss counting for envs that **do** have web workers but **don't** have service workers, but i think that is low risk.

Reply:
Yes, didn't see much merit in separating them instead of treating support as a whole, despite different flavors.

Reply:
Originally intended to just treat them together instead of separately, but now that I see Safari doesn't have SharedWorker support until the latest v16.. maybe they ought to be separated?

Reply:
Didnt realize the coverage was that late, probably should be separated.

Reply:
Done